internal/statemachine/distro.go

```diff
@@ -0,0 +1,46 @@
+package statemachine
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+// releaseFromCodename returns the release version associated to a codename
+func releaseFromCodename(codename string) (string, error) {
+	cmd := exec.Command("ubuntu-distro-info", "-r", "--series", codename)
+	outputBytes, err := cmd.Output() // nolint: errcheck
+	if err != nil {
+		return "", fmt.Errorf("unable to get the release from the codename %s", codename)
+	}
+	// Remove the "LTS" suffix if needed
+	return strings.TrimSpace(strings.Split(string(outputBytes), " ")[0]), nil
+}
+
+// isSeriesEqualOrOlder returns true if a is equal to or older than b
+func isSeriesEqualOrOlder(a, b string) (bool, error) {
+	if a == b {
+		return true, nil
+	}
+	aRelease, err := releaseFromCodename(a)
+	if err != nil {
+		return false, err
+	}
+	bRelease, err := releaseFromCodename(b)
+	if err != nil {
+		return false, err
+	}
+	cmd := exec.Command("dpkg", "--compare-versions", aRelease, "lt", bRelease)
```

Comment:
I think just textual comparison would be OK here, these aren't package versions.

I think the _most_ correct way of doing this would be to compare the locations of the series names in the output of `ubuntu-distro-info --all` but well. This is fine.

Reply:
Your suggestion is much simpler so I reworked it.